k256/src/arithmetic/field/field_5x52.rs

```diff
@@ -487,6 +487,47 @@ impl ConstantTimeEq for FieldElement5x52 {
     }
 }
 
+impl PartialOrd for FieldElement5x52 {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for FieldElement5x52 {
+    fn eq(&self, other: &Self) -> bool {
+        self.ct_eq(other).into()
+    }
+}
+
+impl Eq for FieldElement5x52 {}
+
+/// Returns sign of left - right
+fn cmp_vartime(left: &[u64; 5], right: &[u64; 5]) -> i32 {
+    use core::cmp::Ordering::*;
+
+    // since we're little-endian, need to reverse iterate
+    for (l, r) in left.iter().rev().zip(right.iter().rev()) {
+        match l.cmp(r) {
```

Comment:
I don't understand this fn. Is there a reason is doesn't return an Ordering variant?

Reply:
I believe this is done as a parallel to https://github.com/RustCrypto/elliptic-curves/blob/master/p256/src/arithmetic/scalar.rs#L237 and then it gets invoked from the Ord impl, I don't fully understand the reason to do so tbh.

cc: @madninja

Reply:
I just reused the same code that was used elsewhere in the same repo.

Reply:
Good enough for me. The code looks like it had something to do with trying to defeat the optimizer or something like that